src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -384,7 +384,7 @@ static bool areEqual(SyntaxReference a, SyntaxReference b)
 
         public override int GetHashCode()
         {
-            return syntaxReferenceOpt.GetHashCode();
+            return Syntax.GetHashCode();
```

Comment:
>Syntax [](http://example.com/codeflow?start=19&length=6)

Thanks for catching this. 

This looks distinct from `Equals()` which is checking the fields of `syntaxReferenceOpt`. Should `Equals()` be reverted to check `lambda.Syntax == Syntax` instead of checking `syntaxReferenceOpt`?

Reply:
@cston I *think* both should work the same. I'll go with `lambda.Syntax == Syntax`. It's more clear, and is the original behavior before the lambda improvements work.

https://github.com/dotnet/roslyn/blob/446399422d1d8f7a97a2c04eea8775d3559d206d/src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs#L388